docs/DeveloperGuide.md

```diff
@@ -2,33 +2,106 @@
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+The app is a single-user application that can computerize the processes of making online booking and purchase of movie tickets, and listing of movie details and reviews.
+The app will help customers to make a movie schedule ahead of time, and enable a fast and intuitive ticket purchasing procedure.
+The app will help cinema administrators to update the movie information regularly through an intuitive interface.
 
 ## User Stories
 
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+|v1.0|new customer|follow through with the steps provided|make a booking.|
+|v1.0|indecisive customer|display summaries of several movies|decide which I want to book for.|
+|v1.0|customer|I can have a “quit” command|close the application easily.|
+|v1.0|customer|load the application without any errors|use the other functions.|
+|v1.0|customer|see the location of the cinemas of the particular movie|go to the nearest cinema.|
+|v1.0|customer|see the list of the movie and timing|make plans ahead.|
+|v1.0|customer|book as many seats as it can hold|help my friends book too.|
+|v1.0|customer|book as many movies available|make plans ahead without being restricted.|
+|v2.0|customer|cancel any of my bookings|don't waste my money for being unable to attend.|
+|v2.0|admin|add new movies|make the application is continuously updated.|
+|v2.0|admin|edit new movies|make the application is continuously updated.|
+|v2.0|admin|delete new movies|make the application is continuously updated.|
+|v2.0|customer|filter movies by their rating.|make my decision to either watch them or not.|
+|v2.0|customer|add a few reviews of the movies|convey my opinions about the movie to other watchers.|
+|v2.0|customer|view the reviews of the movies|make my decision to either watch them or not.|
+|v2.0|customer|filter the movies by type, director, actors...|quickly find the movie I like.|
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+1. Usability Requirements
+    1. All text in the system must be consistently written in English.
+    2. The wording used in system text must be intuitive for user to understand easily.   
+    3. The layout of the application must be consistent throughout all the pages.
+    4. The system will provide a list of options for the user to choose.
+2. Performance Requirements
+    1. Database must be regularly updated according to user inputs.
+    2. Exceptions must be performed in cases of undesirable incidents.
+3. Security Requirements
+    1. All passwords must be hashed before it is stored.
+
+## Implementation
```

Comment:
There should be more features of this system to be added in this section. Meanwhile, some examples showing the input and output of the feature would be nice.